In [1]:

from sklearnex import patch_sklearn

patch_sklearn()

from drowsiness_detection.data import get_data_not_splitted, get_identifier_array_not_splitted
from matplotlib import rcParams
from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LinearRegression

rcParams["figure.figsize"] = 10, 10

Intel(R) Extension for Scikit-learn* enabled (https://github.com/intel/scikit-learn-intelex)


In [2]:
X, y = get_data_not_splitted()
ids = get_identifier_array_not_splitted()
print(X.shape, y.shape, ids.shape)

X = X[:10000]
y = y[:10000]
ids = ids[:10000]

(48936, 67) (48936,) (48936, 2)


In [3]:
model = LinearRegression()

In [4]:
%%timeit
model.fit(X, y)

1.38 ms ± 94.9 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)


## Results
#### sklearn execution times for vanilla Logistic Regression:
* 31.3 ms ± 7.3 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
#### same setup but patched with sklearn extensions:
* 1.38 ms ± 94.9 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)